keras_metrics/metrics.py

```diff
@@ -1,20 +1,43 @@
+from functools import partial
 from keras import backend as K
 from keras.layers import Layer
 from operator import truediv
 
 
-def _int32(y_true, y_pred):
-    y_true = K.cast(y_true, "int32")
-    y_pred = K.cast(K.round(y_pred), "int32")
-    
-    if y_pred.shape[1] != 1:
-      y_pred = y_pred[:,1:2]
-      y_true = y_true[:,1:2]
-      
-    return y_true, y_pred
+class layer(Layer):
 
+    def __init__(self, label=None, **kwargs):
+        super(layer, self).__init__(**kwargs)
 
-class true_positive(Layer):
+        # If layer metric is explicitly created to evaluate specified class,
+        # then use a binary transformation of the output arrays, otherwise
+        # calculate an "overal" metric.
+        if label:
+            self.cast_strategy = partial(self._binary, label=label)
+        else:
+            self.cast_strategy = self._categorical
+
+    def cast(self, y_true, y_pred, dtype="int32"):
+        """Convert the specified true and predicted output to the specified
+        destination type (int32 by default).
+        """
+        return self.cast_strategy(y_true, y_pred, dtype=dtype)
+
+    def _binary(self, y_true, y_pred, dtype, label=0):
+        column = slice(label, label+1)
+
+        # Slice a column of the output array.
+        y_true = y_true[:,column]
+        y_pred = y_pred[:,column]
+        return self._categorical(y_true, y_pred, dtype)
+
+    def _categorical(self, y_true, y_pred, dtype):
+        y_true = K.cast(y_true, dtype)
+        y_pred = K.cast(K.round(y_pred), dtype)
+        return y_true, y_pred
+
```

Comment:
Can you check the number of layer here inside the method, if it is bigger than 2, error should be given to user to make him/her choose layer